central/config/datastore/singleton.go

```diff
@@ -48,6 +50,7 @@ var (
 			},
 		},
 		ExpiredVulnReqRetentionDurationDays: DefaultExpiredVulnReqRetention,
+		DefaultReportRetentionWindow:        DefaultReportRetentionWindow,
```

Comment:
I think there's a potential issue here.

On line 74, if there is no config in the database already, then it creates a new one with the default you provided. So for new installations, that works great.

However, for an existing installation (one which already has a config), it _won't_ set a default. Since this is an int I think it'll just return 0. I doubt we want a 0 day retention policy for downloadable reports.

I can think of two ways around it. You can add a check that sees if retention is set to 0, and if so, set it to default. But this means a user cannot set their retention to 0 manually.

Alternatively, change the proto such that the `PrivateConfig` object has a sub object for reports config which has the int inside of it. Look at `DecommissionedClusterRetentionConfig` as an example. Then you can check if the sub-object is null (example in line 79) which allows the user to manually set 0.

Reply:
Second the issue Nakul surfaced.

> Alternatively, change the proto such that the PrivateConfig object has a sub object for reports config which has the int inside of it. Look at DecommissionedClusterRetentionConfig as an example. Then you can check if the sub-object is null (example in line 79) which allows the user to manually set 0.

I would recommend doing the above (otherwise, you must write a migration, which is obviously the hard way).